framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -1524,6 +1521,112 @@ protected void performFlowAnalysisForClass(ClassTree classTree) {
     }
   }
 
+  /**
+   * A helper method for {@link #performFlowAnalysisForClass(ClassTree)} that analyzes {@code
+   * method} and all lambdas contained within it.
+   *
+   * @param classTree class tree containing {@code method}
+   * @param method method to analyze
+   * @param classQueue classes found within {@code method} are added to this queue
+   * @param fieldValues values of fields to be used
+   * @param capturedStore the input Store to use for captured variables, e.g. in a lambda
+   */
+  private void performFlowAnalysisForMethod(
+      ClassTree classTree,
+      CFGMethod method,
+      Queue<IPair<ClassTree, @Nullable Store>> classQueue,
+      List<FieldInitialValue<Value>> fieldValues,
+      @Nullable Store capturedStore) {
+    // The list contains one element for each `return` statement in the lambda (the map key).
+    Map<LambdaExpressionTree, List<AnnotationMirrorSet>> lambdaResultTypeMap = new HashMap<>();
+    Map<LambdaExpressionTree, ControlFlowGraph> lambdaToCFG = new HashMap<>();
+    ControlFlowGraph methodCFG = null;
+
+    boolean isConstructor = TreeUtils.isConstructor(method.getMethod());
+
+    // Analyze `method` and all lambdas contained in `method` until the type of the lambda result
+    // expressions do not change.
+    while (true) {
+      Queue<IPair<ClassTree, @Nullable Store>> classQueueInMethod = new ArrayDeque<>();
+      Queue<IPair<LambdaExpressionTree, @Nullable Store>> lambdaQueueInMethod = new ArrayDeque<>();
+      methodCFG =
+          analyze(
+              classQueueInMethod,
+              lambdaQueueInMethod,
+              method,
+              fieldValues,
+              methodCFG,
+              isConstructor,
+              /* updateInitializationStore= */ false,
+              /* isStatic= */ false,
+              capturedStore);
+
+      boolean anyLambdaResultChanged = false;
+      while (!lambdaQueueInMethod.isEmpty()) {
+        IPair<LambdaExpressionTree, @Nullable Store> lambdaPair = lambdaQueueInMethod.remove();
+        LambdaExpressionTree lambda = lambdaPair.first;
+        MethodTree mt =
```

Comment:
What is the relationship between this and formal parameter `method`?  Can it be cached rather than recomputed on each iteration of the `while` loop?